util/election.go

```diff
@@ -0,0 +1,12 @@
+package util
+
+// If you are running Prometheus in HA mode where each Prometheus instance sends data to corresponding adapter you probably
+// want to allow writes into the database from only one adapter at the time. We need to elect a leader who can write to
+// the database. If leader goes down, another leader is elected. Look at `lock.go` for an implementation based on PostgreSQL
+// advisory locks. Should be easy to plug in different leader election implementations.
+type Election interface {
```

Comment:
godoc comments should actually start with the name of the thing and describe what it is, so in this case it starts with `Election` and then describe what the interface is for.